test-automation/app/hooks.py

```diff
@@ -0,0 +1,23 @@
+import os
+from contextlib import contextmanager
+
+import logging
+from aloe import around, world
+from selenium.webdriver.remote.remote_connection import LOGGER
+from selenium import webdriver
+
+'''
+this method sets a global variable of type WebDriver. This is the browser.
+@around.all is a global hook. It will be ran once before all scenarios
+After the last scenario is completed (pass or fail) the WebDriver will be closed
+'''
+@around.all
+@contextmanager
+def with_firefox():
+    LOGGER.setLevel(logging.WARNING)
+    driver_path = os.path.dirname(os.path.realpath(__file__)).replace("app", "conf\geckodriver.exe")
```

Comment:
geckodriver.exe seems platform specific. How might these tests be run on Linux/Mac?

Reply:
Geckodriver is platform specific. If we change the executable and edit 
.replace("app", "conf\geckodriver.exe")
to 
.replace("app", "conf\geckodriver")
it will work on Linux or Mac.
Alternatively, I can add all geckodriver versions in the conf folder and pick one programmatically based on the user OS. This will increase the size of the project though because each executable is around 5MB.

Reply:
> Alternatively, I can add all geckodriver versions in the conf folder and pick one programmatically based on the user OS.

That sounds reasonable.

Reply:
OK. I will add support for 32, 64-bit Linux and MacOs.